apps/web/src/services/analytics/useMixpanel.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react'
+import mixpanel from 'mixpanel-browser'
+import { IS_PRODUCTION } from '@/config/constants'
+
+/**
+ * Initializes Mixpanel analytics for the web app.
+ * Should be called once at app startup.
+ */
+const useMixpanel = () => {
+  useEffect(() => {
+    // Only initialize in the browser
+    if (typeof window === 'undefined') return
+    const token = process.env.NEXT_PUBLIC_MIXPANEL_TOKEN
+    if (!token) return
+    mixpanel.init(token, { debug: !IS_PRODUCTION })
+  }, [])
+}
```

Comment:
I suggest to add a feature flag here so we can turn it on or off via our config service. There is a hook called `useHasFeature` to handle this and an enum called `FEATURES` that needs to be extended.

Reply:
Yes, looks good!